tests/cache/index-l1-hot.test.js

```diff
@@ -0,0 +1,183 @@
+// const Redis = require('ioredis-mock');
+const _ = require('lodash');
+const CacheSingleton = require('../../src/cache');
+const { BASE_CONFIG } = require('../../src/cache/generate-config');
+const { cacheGetRequest, cacheSetResponse } = require('../../src/utils/cache-request');
+
+jest.mock('ioredis', () => {
+  // eslint-disable-next-line global-require
+  const Redis = require('ioredis-mock');
+
+  if (typeof Redis === 'object') {
+    // the first mock is an ioredis shim because ioredis-mock depends on it
+    // https://github.com/stipsan/ioredis-mock/blob/master/src/index.js#L101-L111
+    return {
+      Command: { _transformer: { argument: {}, reply: {} } },
+    };
+  }
+
+  // second mock is the actual constructor. note that this must be
+  // a `function` and NOT an anonymous function, as the latter is not a valid
+  // ES constructor
+
+  // eslint-disable-next-line func-names
+  return function (args) {
+    const object = new Redis({ ...args, lazyConnect: false });
+    object.status = 'ready';
+
+    return object;
+  };
+});
```

Comment:
This is a mock that generates a mocked Redis client using `ioredis-mock`. There is a weird bug going on in it that requires this complicated shimming logic.

https://github.com/stipsan/ioredis-mock/issues/568

Reply:
hmmm, interesting